sdks/python/apache_beam/testing/test_stream.py

```diff
@@ -291,10 +291,10 @@ def expand(self, pbegin):
     assert isinstance(pbegin, pvalue.PBegin)
     self.pipeline = pbegin.pipeline
     if not self.output_tags:
-      self.output_tags = set([None])
+      self.output_tags = {None}
```

Comment:
If the user explicitly sets the output tags to {None}, they might be expecting a dict. (Specifically, they might get a set from elsewhere, and set the output tags from that set, and it would be awkward to have to check that set to determine how to interpret the result. So in this case I would do

```
if not self.output_tags:
  return pvalue.PCollection(self.pipeline, is_bounded=False)
else:
  return { ... for tag in self.output_tags}
```

Reply:
This is a little harder to implement, mainly because the TestStream retrieves its output_tags from the keys of the PTransform payload holding it. This means that output_tags = None and output_tags = {None} look the same to the PTransform payload outputs as a map with a single key being None. When a TestStream is reconstructed, even if the original output_tags was unset, it will be constructed with output_tags = {None}.

I think the best we can do is to treat {None} and None the same way.

Reply:
OK, in that case I'm fine with this.